routing/src/vrf.rs

```diff
@@ -0,0 +1,447 @@
+        // So, we have to replace it with a dummy route with action Drop, to actually represent a lack of route.
+        if prefix == &Ipv6Prefix::default() {
+            // This is a bit of a hack
+            if let Some(mut prior) = self.routesv6.insert(*prefix, Route::default()) {
+                self.deregister_shared_nexthops(&mut prior);
+            }
+            self.add_route_v6(prefix, Route::default(), &[RouteNhop::default()]);
+        } else if let Some(found) = &mut self.routesv6.remove(prefix) {
+            self.deregister_shared_nexthops(found);
+        }
+    }
+    pub fn del_route(&mut self, prefix: &Prefix) {
+        match prefix {
+            Prefix::IPV4(p) => self.del_route_v4(p),
+            Prefix::IPV6(p) => self.del_route_v6(p),
+        }
+    }
+
+    /////////////////////////////////////////////////////////////////////////
+    // Route retrieval
+    /////////////////////////////////////////////////////////////////////////
+
+    #[inline(always)]
+    fn get_route_v4(&self, prefix: &Ipv4Prefix) -> Option<&Route> {
+        self.routesv4.get(prefix)
+    }
+    #[inline(always)]
+    fn get_route_v6(&self, prefix: &Ipv6Prefix) -> Option<&Route> {
+        self.routesv6.get(prefix)
+    }
+    pub fn get_route(&self, prefix: &Prefix) -> Option<&Route> {
+        match prefix {
+            Prefix::IPV4(p) => self.get_route_v4(p),
+            Prefix::IPV6(p) => self.get_route_v6(p),
+        }
+    }
+
+    /////////////////////////////////////////////////////////////////////////
+    // Route retrieval (mutable): we may not need this and if we do, extra
+    // care should be taken modifying route internals
+    /////////////////////////////////////////////////////////////////////////
+
+    #[inline(always)]
+    fn get_route_v4_mut(&mut self, prefix: &Ipv4Prefix) -> Option<&mut Route> {
+        self.routesv4.get_mut(prefix)
+    }
+    #[inline(always)]
+    fn get_route_v6_mut(&mut self, prefix: &Ipv6Prefix) -> Option<&mut Route> {
+        self.routesv6.get_mut(prefix)
+    }
+    pub fn get_route_mut(&mut self, prefix: &Prefix) -> Option<&mut Route> {
+        match prefix {
+            Prefix::IPV4(p) => self.get_route_v4_mut(p),
+            Prefix::IPV6(p) => self.get_route_v6_mut(p),
+        }
+    }
+
+    /////////////////////////////////////////////////////////////////////////
+    // LPM, single call
+    /////////////////////////////////////////////////////////////////////////
+
+    #[inline(always)]
+    fn lpm_v4(&self, target: &Ipv4Prefix) -> (&Ipv4Prefix, &Route) {
+        self.routesv4.lookup(target)
+    }
+    #[inline(always)]
+    fn lpm_v6(&self, target: &Ipv6Prefix) -> (&Ipv6Prefix, &Route) {
+        self.routesv6.lookup(target)
+    }
+    pub fn lpm(&self, target: &IpAddr) -> (Prefix, &Route) {
+        match *target {
+            IpAddr::V4(a) => {
+                let (p, r) = self.lpm_v4(&Ipv4Prefix::from(a));
+                (Prefix::IPV4(*p), r)
```

Comment:
not a big deal, but you should be able to do `self.lpm_v4(&a.into())` here

Reply:
Ack!